src/pyfmi/fmi.pyx

```diff
@@ -496,33 +513,49 @@ cdef class ModelBase:
         for i in range(N):
             print(log[i])
 
-    def extract_xml_log(self, file_name=None):
+    def extract_xml_log(self, file_name=None, stream = None):
         """
         Extract the XML contents of a FMU log and write as a new file.
 
         Parameters::
 
             file_name --
-                Name of the file which holds the extracted log
+                Name of the file which holds the extracted log, or a stream to write to
+                that supports the function 'write'. Default behaviour is to write to a file.
                 Default: get_log_filename() + xml
+            stream --
```

Comment:
Hmm I see, what is the use-case here? We already have the stream where we've stored the data?

Reply:
Yes correct, see my comment above where I wrote an example with 3 streams, if it doesn't make sense I can change it but that is the reason why I added it.